ui/src/app/pages/access/access-renewal-page.tsx

```diff
@@ -52,50 +55,103 @@ const renewalStyle = {
     width: 560
   }
 };
+const isInGoodStanding = (nextReview: number): boolean => daysFromNow(nextReview) > LOOKBACK_PERIOD;
 
-export const withInvalidDateHandling = date => {
+const withInvalidDateHandling = date => {
   if (!date) {
     return 'Unavailable';
   } else {
     return displayDateWithoutHours(date);
   }
 };
 
+const computeDisplayDates = (lastConfirmedTime, bypassTime, nextReviewTime) => {
+  const userCompletedModule = !!lastConfirmedTime;
+  const userBypassedModule = !!bypassTime;
+  const lastConfirmedDate = withInvalidDateHandling(lastConfirmedTime);
+  const nextReviewDate = withInvalidDateHandling(nextReviewTime);
+  const bypassDate = withInvalidDateHandling(bypassTime);
+
+  return cond(
+    // User has bypassed module
+    [userBypassedModule, () => ({lastConfirmedDate: `${bypassDate}`, nextReviewDate: 'Unavailable (bypassed)'})],
+    // User never completed training
+    [!userCompletedModule && !userBypassedModule, () =>
+      ({lastConfirmedDate: 'Unavailable (not completed)', nextReviewDate: 'Unavailable (not completed)'})],
+    // User completed training, but is in the lookback window
+    [userCompletedModule && !isInGoodStanding(nextReviewTime), () => {
```

Comment:
I don't think "isInGoodStanding" is quite right semantically, for users who are not yet expired.  maybe invert this to isExpiringSoon or similar?

Reply:
also: these two conditions can be collapsed

Reply:
On collapsing the two we would lose the `(expired)` string - or were you referring to something else? Merging as-is for now - feel free to follow up with a comment, I can integrate a change in the next PR.

Reply:
We can keep the (expired) logic for both cases, since it will never be true for `isCompleted = true`.  But it's also fine as-is.